### PR TITLE
item_db and item_combo_db correction #7

### PR DESCRIPTION
**Pull request by kaninhot004**
* **Server Mode**: 

Renewal

* **Description of Pull Request**: 

More of item_db and item_combo_db corrections.

https://www.divine-pride.net/database/item/28551/
https://www.divine-pride.net/database/item/28372/
https://www.divine-pride.net/database/item/1837/
https://www.divine-pride.net/database/item/15040/
https://www.divine-pride.net/database/item/18676/
https://www.divine-pride.net/database/item/22042/
https://www.divine-pride.net/database/item/13088/
https://www.divine-pride.net/database/item/28372/
https://www.divine-pride.net/database/item/18864/
https://www.divine-pride.net/database/item/18934/
https://www.divine-pride.net/database/item/18938/
https://www.divine-pride.net/database/item/28602/
https://www.divine-pride.net/database/item/2196/
https://www.divine-pride.net/database/item/28342/
https://www.divine-pride.net/database/item/2597/
https://www.divine-pride.net/database/item/19031/
https://www.divine-pride.net/database/item/22064/
https://www.divine-pride.net/database/item/2124/
https://www.divine-pride.net/database/item/2702/
https://www.divine-pride.net/database/item/15145/
https://www.divine-pride.net/database/item/18652/
https://www.divine-pride.net/database/item/19048/
https://www.divine-pride.net/database/item/19019/

Thanks to Jumbo # 8053 (Discord)

### CODE REVIEW

**Review by crazy-arashi — CHANGES_REQUESTED** [review]

File: db/re/item_combo_db.txt [hunk 1]
```diff
@@ -762,8 +762,8 @@
 18776:20710,{ bonus bBaseAtk,10; }
 18776:22015,{ bonus bMatk,20; }
 18823:19246,{ .@atk = 40; .@aspd = 3; .@dmg = 2; .@agi = readparam(bAgi); if (.@agi > 107) { .@atk += 60; .@aspd += 5; .@dmg += 2; } if (.@agi > 119) { .@atk += 80; .@aspd += 7; .@dmg += 4; } bonus bBaseAtk,.@atk; bonus bAspdRate,.@aspd; bonus2 bSubRace,RC_Player_Human,.@dmg; bonus2 bResEff,Eff_Blind,10000; bonus2 bResEff,Eff_Silence,10000; }
-18823:28372,{ bonus2 bSkillVariableCast,"CR_GRANDCROSS",-1500; bonus2 bSkillAtk,"LG_RAYOFGENESIS",BaseLevel/30 + BaseLevel; bonus2 bSkillUseSP,"LG_RAYOFGENESIS",-10; }
-18823:28551,{ bonus2 bSkillDelay,"LG_OVERBRAND",-2000; bonus bLongAtkRate,getskilllv("LG_CANNONSPEAR") * 2; bonus bLongAtkRate,getskilllv("LG_OVERBRAND") * 2; bonus2 bSkillAtk,"LG_CANNONSPEAR",30; bonus2 bSkillAtk,"LG_OVERBRAND",20; }
+18823:28372,{ bonus2 bSkillVariableCast,"CR_GRANDCROSS",-2000; bonus2 bSkillAtk,"LG_RAYOFGENESIS",BaseLevel/30; bonus2 bSkillUseSP,"LG_RAYOFGENESIS",-10; bonus2 bSkillAtk,"CR_GRANDCROSS",BaseLevel; }
```
[the remark this answers is not in the document]
Comment (kaninhot004) [reply db/re/item_combo_db.txt:765]: image
In Divine Pride I had selected kRO servers, And translated item description by Google it said -2 seconds.

Do you had any description for this? Thanks!

File: db/re/item_combo_db.txt [hunk 2]
```diff
@@ -762,8 +762,8 @@
...
+18823:28551,{ bonus bLongAtkRate,getskilllv("LG_CANNONSPEAR")*2; bonus bLongAtkRate,getskilllv("LG_OVERBRAND")*2; bonus2 bSkillAtk,"LG_CANNONSPEAR",20; bonus2 bSkillAtk,"LG_OVERBRAND",20; }
```
[the remark this answers is not in the document]
Comment (kaninhot004) [reply db/re/item_combo_db.txt:766]: This is from Divine Pride kRO server selected, Please check it for me again.
If I wrong I will use your changes. Thanks!

image

File: db/re/item_db.txt [hunk 3]
```diff
@@ -1174,7 +1174,7 @@
 1834,TE_Woe_Fist,TE Woe Fist,5,0,,0,150,,1,0,0x00008100,63,2,2,3,40,1,12,{ bonus2 bAddRace,RC_Player_Human,40; bonus2 bAddEff,Eff_Freeze,3000; },{},{}
 1835,Spartacus,Spartacus,5,0,,600,100,,1,2,0x00008100,63,2,2,4,50,1,12,{ bonus bUnbreakableWeapon; .@r = getrefine(); bonus2 bAddClass,Class_All,.@r; if(.@r>9) { bonus bNoSizeFix; } },{},{}
 1836,Thanos_Knuckle,Thanos Knuckle,5,10,,600,160:100,,1,1,0x00008000,56,2,2,4,120,1,12,{ bonus bInt,6; bonus bVit,6; bonus bLuk,-6; bonus2 bHPDrainRate,50,5; bonus2 bSPDrainRate,10,5; bonus2 bHPLossRate,100,10000; },{},{ heal -1000,0; }
-1837,Iron_nail,Iron Nail,5,10,,1500,100,,,1,0x00008100,63,2,2,3,50,1,12,{ bonus bAspdRate,-5; bonus bAspdRate,((getrefine()+1)/2); },{},{}
+1837,Iron_nail,Iron Nail,5,10,,1500,100,,,1,0x00008100,63,2,2,3,50,1,12,{ bonus bIgnoreDefClass,Class_All; .@r = (getrefine()/2); bonus bAspdRate,-5+.@r; if(getskilllv("MO_CHAINCOMBO")==5){ bonus bBaseAtk,20; } if(getskilllv("MO_COMBOFINISH")==5){ bonus bBaseAtk,20; } if(getskilllv("CH_TIGERFIST")==5){ bonus bBaseAtk,20; } if(getskilllv("CH_CHAINCRUSH")==5){ bonus bBaseAtk,40; } },{},{}
```
[the remark this answers is not in the document]
Comment (kaninhot004) [reply db/re/item_db.txt:1177]: There are no description about Unbreakable Weapon, Please help me check it again. Thanks!

image

[the remark this answers is not in the document]
Comment (Atemo) [reply db/re/item_db.txt:1177]: Select jRO server, some items can't be found in the itemInfo lua Korean
`絶対に壊れない ` (never break) in 1837 description

**Comment by saya9200** [comment]
I really appreciate the item additions @kaninhot004 , I've been looking for some of them!

**Review by crazy-arashi — APPROVED** [review]

**Comment by zdurexz** [comment]
@Atemo Please check

File: db/re/item_db.txt [hunk 4]
```diff
@@ -10066,10 +10066,10 @@
...
 18932,Rhino_Model_Hat,Rhino Model Hat,4,10,,300,,10,,1,0xFFFFFFFF,63,2,256,,30,1,1068,{ .@r = getrefine(); bonus bVit,2+((.@r >= 11) ? (.@r/11) : 0); bonus3 bAutoSpellWhenHit,"WZ_VERMILION",1,50; },{},{}
 18933,Mechanical_Plant_Hat,Mechanical Plant Hat,4,20,,500,,10,,1,0xFFFFFFFF,63,2,256,,45,1,1069,{ .@r = getrefine()/2; bonus bDex,2; bonus bMdef,7; bonus2 bSkillAtk,"GN_SPORE_EXPLOSION",5+(5*.@r); },{},{}
-18934,Fox_Ears_Bell_Ribbon,Fox Ears Bell Ribbon,4,0,,400,,4,,1,0xFFFFFFFF,63,2,256,,50,1,1070,{ bonus bAgi,2; bonus bAspdRate,10; .@r = getrefine(); autobonus "{ .@r = getrefine(); bonus bCritical,100; bonus bLongAtkRate,5+(.@r > 6 ? .@r : 0); }",50+(.@r*2),5000,BF_NORMAL,"{ active_transform 1150,5000; }"; },{},{}
+18934,Fox_Ears_Bell_Ribbon,Fox Ears Bell Ribbon,4,0,,400,,4,,1,0xFFFFFFFF,63,2,256,,50,1,1070,{ bonus bAgi,2; bonus bAspdRate,10; .@r = getrefine(); autobonus "{ .@i = .@r-6; bonus bCritical,100; bonus bLongAtkRate,5+(.@r > 6 ? .@i : 0); }",50+(.@r*2),5000,BF_NORMAL,"{ active_transform 1150,5000; }"; },{},{}
```
Comment (Atemo) [inline db/re/item_db.txt:10069]: The string in autobonus is read independently of the item script, `getrefine()` must be redefined inside
```suggestion
18934,Fox_Ears_Bell_Ribbon,Fox Ears Bell Ribbon,4,0,,400,,4,,1,0xFFFFFFFF,63,2,256,,50,1,1070,{ bonus bAgi,2; bonus bAspdRate,10; .@r = getrefine(); autobonus "{ bonus bCritical,100; bonus bLongAtkRate,5+max(0,getrefine()-6); }",50+(.@r*2),5000,BF_NORMAL,"{ active_transform 1150,5000; }"; },{},{}
```

**Review by Atemo — CHANGES_REQUESTED** [review]

**Comment by kaninhot004** [comment]
All changes are done. Thank you!

**Review by Atemo — APPROVED** [review]

**Comment by Atemo** [comment]
Thanks !

**Pull request merged**